Adding new VSCode extensions

superherointj opened the pull request:
* It respects alphabetical order so data can be normalized.

List of changes:
CoenraadS.bracket-pair-colorizer: init at 1.0.61
donjayamanne.githistory: init at 0.6.14
esbenp.prettier-vscode: init at 5.8.0
file-icons.file-icons: init at 1.0.28
freebroccolo.reasonml: init at 1.0.38
jnoortheen.nix-ide: updated metadata
ocamllabs.ocaml-platform: init at 1.5.1

hugolgst commented:
Result of `nixpkgs-review pr 109120` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>vscode-extensions.CoenraadS.bracket-pair-colorizer</li>
    <li>vscode-extensions.donjayamanne.githistory</li>
    <li>vscode-extensions.esbenp.prettier-vscode</li>
    <li>vscode-extensions.file-icons.file-icons</li>
    <li>vscode-extensions.freebroccolo.reasonml</li>
    <li>vscode-extensions.ocamllabs.ocaml-platform</li>
  </ul>
</details>

SuperSandro2000 commented on pkgs/misc/vscode-extensions/default.nix:
```diff
@@ -87,6 +87,40 @@ let
+      };
+
+      CoenraadS.bracket-pair-colorizer = buildVscodeMarketplaceExtension {
+        meta = with stdenv.lib; {
```
```suggestion
        meta = with lib; {
```

SuperSandro2000 commented on pkgs/misc/vscode-extensions/default.nix:
```diff
@@ -119,6 +153,40 @@ let
       };
 
+      esbenp.prettier-vscode = buildVscodeMarketplaceExtension {
+        meta = with stdenv.lib; {
```
```suggestion
        meta = with lib; {
```

SuperSandro2000 commented on pkgs/misc/vscode-extensions/default.nix:
```diff
@@ -119,6 +153,40 @@ let
+      };
+
+      file-icons.file-icons = buildVscodeMarketplaceExtension {
+        meta = with stdenv.lib; {
```
```suggestion
        meta = with lib; {
```

SuperSandro2000 commented on pkgs/misc/vscode-extensions/default.nix:
```diff
@@ -143,6 +211,23 @@ let
       };
 
+      freebroccolo.reasonml = buildVscodeMarketplaceExtension {
+        meta = with stdenv.lib; {
```
```suggestion
        meta = with lib; {
```

SuperSandro2000 commented on pkgs/misc/vscode-extensions/default.nix:
```diff
@@ -192,15 +277,20 @@ let
       };
 
       jnoortheen.nix-ide = buildVscodeMarketplaceExtension {
+        meta = with stdenv.lib; {
```
```suggestion
        meta = with lib; {
```

superherointj replied:
Thanks. Just added changed in batch. After PR is accepted, I'll submit "pkgs/misc/vscode-extensions/default.nix" normalized (alphabetical order, following same standard). Haven't done so yet because I don't want to mix with these new extensions. I was already working on this before "treewide: with stdenv.lib; in meta -> with lib;" change.

SuperSandro2000 commented on pkgs/misc/vscode-extensions/default.nix:
```diff
@@ -283,6 +373,23 @@ let
       };
 
+      ocamllabs.ocaml-platform = buildVscodeMarketplaceExtension {
+        meta = with stdenv.lib; {
```
```suggestion
        meta = with lib; {
```

superherointj commented:
For a new PR (which is ready), I have normalized "pkgs/misc/vscode-extensions/default.nix". Refactored to respect alphabetical order, standardized meta-data, filled meta-data for currently available extensions. Removed `stdenv.lib`. I need this PR to be approved before I can push the new PR so I do not mix new extensions and refactoring. Would appreciate if anyone can merge this. Thanks.

SuperSandro2000 commented on pkgs/misc/vscode-extensions/default.nix:
```diff
@@ -87,6 +87,40 @@ let
       };
 
+      donjayamanne.githistory = buildVscodeMarketplaceExtension {
+        meta = with lib; {
```
Normally meta is at the end of the package definition. I think I would like to maintain this here. Feels a bit nity but it should make treewide changes easier because you can assume a structure.

superherointj replied:
This is an important topic you bring on. This has come up before. Currently VSCode extension list is unordered. And deciding how it should be standardized and enforced is very relevant to allow the extensions list to grow orderly. I agree personal preferences can be very subjective with each person having their own and because of it would be very nice having a standard and following it.

I do not know or I do not understand what "treewide changes" mean. Can you elaborate more about it so I can better understand it?

Alphabetical order is usually a widely acceptable standard that does not require much explanation and it was my initial expectation of this being the standard method of ordering keys in sets because sets are ordered by keys in alphabetical order. Subjectively speaking, having metadata at top also did seem friendlier to me. "documentation" coming before "implementation" seems fine enough. It could be my conditioning at play. Alphabetical ordering seems easier/natural to me than other ordering methods. But important thing is that we agree on some ordering.

I expect VSCode extensions list to be eventually automated (like in #10858), so in the end it will likely follow an alphabetical order. I have normalized the whole list of extensions by alphabetical order and was waiting this PR to be merged before doing PR to normalize the extension list. But as you have some other preference than alphabetical order I'll scrap it.

jonringer replied:
eventually I would like for the vscode plugins to relfect vim plugins, and just have an occasional `./update.sh` ran to bump them.

SuperSandro2000 replied:
> I do not know or I do not understand what "treewide changes" mean. Can you elaborate more about it so I can better understand it?

If we want to change something about all vscode extensions for example. I think we should just ignore this for now and if we convert it to a generator which is a way better idea here, then the problem just goes away.

SuperSandro2000 commented:
@superherointj can you please fix the merge conflict?

superherointj committed:
maintainers: add superherointj

superherointj committed:
vscode-extensions.CoenraadS.bracket-pair-colorizer: init at 1.0.61

superherointj committed:
vscode-extensions.donjayamanne.githistory: init at 0.6.14

superherointj committed:
vscode-extensions.esbenp.prettier-vscode: init at 5.8.0

superherointj committed:
vscode-extensions.file-icons.file-icons: init at 1.0.28

superherointj committed:
vscode-extensions.freebroccolo.reasonml: init at 1.0.38

superherointj committed:
vscode-extensions.jnoortheen.nix-ide: updated metadata

superherointj committed:
vscode-extensions.ocamllabs.ocaml-platform: init at 1.5.1

superherointj commented:
Is it fine now or do I have to change order or do something else?

jonringer reviewed (approved):
LGTM

```
https://github.com/NixOS/nixpkgs/pull/109120

6 packages built:
vscode-extensions.CoenraadS.bracket-pair-colorizer vscode-extensions.donjayamanne.githistory vscode-extensions.esbenp.prettier-vscode vscode-extensions.file-icons.file-icons vscode-extensions.freebroccolo.reasonml vscode-extensions.ocamllabs.ocaml-platform
```

thanks @SuperSandro2000
